Backend Plugins: Collect and expose metrics and plugin process health check

marefr opened the pull request:
**What this PR does / why we need it**:
Use changes from grafana/grafana-plugin-sdk-go#32

Adds support for collecting metrics from backend plugins and exposing them thru Grafana's Prometheus metrics endpoint. The following metrics are included automatically:
![image](https://user-images.githubusercontent.com/1668778/72333029-689b1680-36bb-11ea-9912-91f6b13929f8.png)

Need to decide to expose camelCase or snake_case labels, currently snake_case are used.

Closes #20984

**Special notes for your reviewer**:

marefr committed:
add support for collecting metrics

marefr committed:
use diagnostics_refactor branch of sdk

marefr committed:
add plugin build info metric for all non-core plugin

marefr committed:
Merge remote-tracking branch 'origin/master' into plugin_diagnostics_refactor

marefr committed:
Merge remote-tracking branch 'origin/master' into plugin_diagnostics_refactor

marefr committed:
change plugin metric labels to snake case

ryantxu commented on pkg/plugins/backendplugin/backend_plugin.go:
```diff
@@ -96,3 +119,163 @@ func (p *BackendPlugin) stop() error {
+
+// convertMetricFamily converts metric family to prometheus.Metric.
+// Copied from https://github.com/prometheus/node_exporter/blob/3ddc82c2d8d11eec53ed5faa8db969a1bb81f8bb/collector/textfile.go#L66-L165
+func convertMetricFamily(pluginID string, metricFamily *dto.MetricFamily, ch chan<- prometheus.Metric, logger log.Logger) {
```
This may be a totally fine approach.... so ignore me if I am crazy...

Arent the exports all simple text?  Could we just export text and then search+replace a namespaced prefix?  I don't have a sense of if that is more or less fragil than trying to reconstruct the metrics in the local collector

marefr replied:
We need to convert metrics to a contract that prometheus golang collector expects.

marefr committed:
register and implement backend plugin health check http handler

Enables to check health of backend plugin by using the route
`/api/plugins/<plugin id>/health`.

marefr committed:
change logger name for plugin collector

marefr committed:
Merge remote-tracking branch 'origin/master' into plugin_diagnostics_refactor

kylebrandt reviewed (approved):
This is look good to me. Before merging, we will want to merge the corresponding sdk branch - tag it, and change the import here to be that tag.

marefr committed:
use sdk v0.6.0
